src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -150,6 +151,21 @@ public ResultResource(ProgrammingExerciseParticipationService programmingExercis
         }
         // TODO: How can we catch a more specific exception here? Because of the adapter pattern this is always just Exception...
         catch (Exception ex) {
+            if (ex instanceof BitbucketException) {
```

Comment:
I would suggest catching all of the exceptions in separate catch clause instead of calling instanceof. The IndexOutOfBounds and JenkinsExceptions can be bundled since their catch bodies are identical.
My suggestion:
```java
catch (BitbucketException ex) {
            log.error("Exception encountered when trying to retrieve the plan key from a request a new programming exercise result: {}, {} :"
                + "BitBucket encountered an Exception while trying to retrieve the build plan ", ex, requestBody);
            return badRequest();
} catch (IndexOutOfBoundsException | JenkinsException ex) {
            log.error("Exception encountered when trying to retrieve the plan key from a request a new programming exercise result: {}, {}:"
                + " The full name of the Jenkins build plan was malformed!", ex, requestBody);
            return badRequest();
} catch (Exception ex) {
            log.error("Exception encountered when trying to retrieve the plan key from a request a new programming exercise result: {}, {}", ex, requestBody);
            return badRequest();
}
```

Reply:
I think the exception-handling in `continuousIntegrationService.get().getPlanKey(requestBody);` should rather be fixed instead of catching each possible type. 

The service itself should only throw a `ContinuousIntegrationException` which is correctly implemented in BambooService but not in the JenkinsService

Reply:
Should this be fixed as part of this PR?

Reply:
I agree with @ivanchimeno. Making sure that the JenkinsService throws a `ContinuousIntegrationException` makes more sense. The fix should be rather simple, because the code in the `JenkinsService `only has to be wrapped in a try catch block and the `getPlan` method is only called by `ResultResource`. I would suggest fixing it in this ticket